src/main/java/com/example/petbuddybackend/utils/exception/advice/GeneralAdvice.java

```diff
@@ -138,4 +138,10 @@ public ApiExceptionResponse handleDateRangeException(DateRangeException e) {
         return new ApiExceptionResponse(e, e.getMessage());
     }
 
+    @ExceptionHandler(IllegalArgumentException.class)
+    @ResponseStatus(code = HttpStatus.BAD_REQUEST)
+    public ApiExceptionResponse handleIllegalArgumentException(IllegalArgumentException e) {
+        return new ApiExceptionResponse(e, e.getMessage());
+    }
+
```

Comment:
IllegalArgumentException bym nie ruszał bo to typowo javovy wyjątek i może się rzucić w losowych miejscach bibliotek których używamy, więc imo zostawiłbym to żeby zwracało 500.

Ten wyjątek powinien być rzucany jeżeli błąd jest po stronie programisty a nie błędnego requestu. Na przykład ten wyjątek ci się rzuci jak zrobisz

`new ArrayList<>(-1);`

Reply:
dobra, to wymienie w serwisie na inny wyjatek.